src/matches.rs

```diff
@@ -135,9 +135,8 @@ impl LateLintPass for MatchPass {
 }
 
 fn check_single_match(cx: &LateContext, ex: &Expr, arms: &[Arm], expr: &Expr) {
-    if arms.len() == 2 &&
-       arms[0].pats.len() == 1 && arms[0].guard.is_none() &&
-       arms[1].pats.len() == 1 && arms[1].guard.is_none() {
+    if arms.len() == 2 && arms[0].pats.len() == 1 && arms[0].guard.is_none() && arms[1].pats.len() == 1 &&
+       arms[1].guard.is_none() {
         let els = if is_unit_expr(&arms[1].body) {
```

Comment:
I prefer the previous version, with `arms[0]` on one line and `arms[1]` on the other.